lib/spack/spack/version.py

```diff
@@ -805,26 +822,32 @@ def __contains__(self, other):
         )
         return in_upper
 
-    @coerced
-    def satisfies(self, other):
-        """
-        x.satisfies(y) in general means that x and y have a
-        non-zero intersection. For VersionRange this means they overlap.
+    def intersects(self, other) -> bool:
+        """Return two if two version ranges overlap with each other, False otherwise.
 
-        `satisfies` is a commutative binary operator, meaning that
-        x.satisfies(y) if and only if y.satisfies(x).
+        This is a commutative operation.
 
-        Note: in some cases we have the keyword x.satisfies(y, strict=True)
-        to mean strict set inclusion, which is not commutative. However, this
-        lacks in VersionRange for unknown reasons.
-
-        Examples
+        Examples:
         - 1:3 satisfies 2:4, as their intersection is 2:3.
         - 1:2 does not satisfy 3:4, as their intersection is empty.
         - 4.5:4.7 satisfies 4.7.2:4.8, as their intersection is 4.7.2:4.7
+
+        Args:
+            other: version range to be checked for intersection
         """
         return self.overlaps(other)
 
+    @coerced
+    def satisfies(self, other):
+        """A version range satisfies another if it is a subset of the other.
+
+        Examples:
+        - 1:2 does not satisfy 3:4, as their intersection is empty.
+        - 1:3 does not satisfy 2:4, as they overlap but neither is a subset of the other
+        - 1:3 satisfies 1:4.
+        """
+        return self.overlaps(other) and self.intersection(other) == self
```

Comment:
Is computing `self.intersection` too expensive to just do `return self.intersection(other) == self`?

Reply:
duh, intersection calls overlaps itself

Reply:
I guess the expensiveness depends on whether this returns true or false more often in practice.

Reply:
Done in https://github.com/spack/spack/pull/35681/commits/76d4ef974669290b9d8026f8bb4fbf29c292dff2